src/cascadia/WindowsTerminal/IslandWindow.cpp

```diff
@@ -308,5 +308,135 @@ void IslandWindow::UpdateTheme(const winrt::Windows::UI::Xaml::ElementTheme& req
     ::InvalidateRect(_window.get(), nullptr, false);
 }
 
+// Method Description:
+// - Toggles our fullscreen state. See _SetIsFullscreen for more details.
+// Arguments:
+// - <none>
+// Return Value:
+// - <none>
+void IslandWindow::ToggleFullscreen()
+{
+    _SetIsFullscreen(!_fullscreen);
+}
+
+// Method Description:
+// - Controls setting us into or out of fullscreen mode. Largely taken from
+//   Window::SetIsFullscreen in conhost.
+// - When entering fullscreen mode, we'll save the current window size and
+//   location, and expand to take the entire monitor size. When leaving, we'll
+//   use that saved size to restore back to.
+// - When we're entering fullscreen we need to do some additional modification
+//   of our window styles. However, the NonClientIslandWindow very explicitly
+//   _doesn't_ ned to do these steps. Subclasses should override
+//   _ShouldUpdateStylesOnFullscreen to disable setting these window styles.
+// Arguments:
+// - fFullscreenEnabled true if we should enable fullscreen mode, false to disable.
+// Return Value:
+// - <none>
+void IslandWindow::_SetIsFullscreen(const bool fFullscreenEnabled)
+{
+    // It is possible to enter _SetIsFullscreen even if we're already in full
+    // screen. Use the old is in fullscreen flag to gate checks that rely on the
+    // current state.
+    bool fOldIsInFullscreen = _fullscreen;
+    _fullscreen = fFullscreenEnabled;
+
+    // Note: The NonClientIslandWindow _doesn't_ need these style modifications,
+    // so it's overriden _ShouldUpdateStylesOnFullscreen to return false. Doing
+    // these modifications to that window will cause a vista-style window frame
+    // to briefly appear when entering and exiting fullscreen.
+    if (_ShouldUpdateStylesOnFullscreen())
+    {
+        HWND const hWnd = GetWindowHandle();
+
+        // First, modify regular window styles as appropriate
+        LONG dwWindowStyle = GetWindowLongW(hWnd, GWL_STYLE);
+        if (_fullscreen)

```

Comment:
Are you aware of `WI_SetFlagIf` and `WI_ClearFlagIf`?

Reply:
I don't think that works here unfortunately, since WS_OVERLAPPEDWINDOW isn't a single flag, and there's no `WI_SetAllFlagsIf`...

Reply:
Aww butts. OK. Nevermind then.